src/Core/Parser/SyntaxTree/Expression/CastingExpressionNode.php

```diff
@@ -113,8 +115,17 @@ protected static function convertToArray($variable)
         if (is_array($variable)) {
             return $variable;
         }
-        if (is_string($variable) && strpos($variable, ',')) {
-            return array_map('trim', explode(',', $variable));
+        if (is_string($variable)) {
+            // rough check on whether or not it's a json string
+            if (preg_match('/^[\[\{].*[\]\}]$/', trim($variable))) {
+                $array = @json_decode($variable, true);
```

Comment:
Any particular reason why you use `@` here? If you wanted to cover this:

> If depth is outside the allowed range, a [ValueError](https://www.php.net/manual/en/class.valueerror.php) is thrown as of PHP 8.0.0, while previously, an error of level E_WARNING was raised.

I would rather add a try/catch here since Fluid requires PHP 8.1 now. This makes it more explicit. But I think an exception would also be fine in this case.

Reply:
Also, do you have any particular reason why you use `JSON_THROW_ON_ERROR` for encode, but not for decode?

Reply:
@s2b, no reason, just thought it might be nice to not let the frontend explode when there's an issue with the provided string. But you're right I guess ;-)

I've changed that and added a test for an invalid JSON too.